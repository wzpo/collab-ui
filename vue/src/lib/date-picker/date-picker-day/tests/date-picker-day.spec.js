import { mount } from '@vue/test-utils';
import DatePickerDay from '../index.vue';
import Button from '../../../button/index.vue';
import Vue from 'vue';
import moment from 'moment-timezone';

Vue.component(Button.name, Button);

const TEST_DATE = '2018-04-01';

describe('DatePickerDay', () => {
  let day;

  beforeAll(() => {
    // setting timezone to CST
    moment.tz.setDefault('Asia/Kolkata');
    day = moment(TEST_DATE);
  });

  afterAll(() => {
    moment.tz.setDefault();
  });

  it('should match snapshot', () => {
    const selected = moment(TEST_DATE);
    const wrapper = mount(DatePickerDay, {
      propsData: {
        day: day,
        month: 3,
      },
      provide: {
        getSelected: () => selected
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('should apply the right modifier classes', () => {
    it('should apply a modifier, when the date does not belong to current month', () => {
      const container = mount(DatePickerDay, {
        propsData: {
          day: day,
          month: 10,
        },
      });
      expect(container.classes('md-datepicker__day--outside-month')).toEqual(true);
    });

    it('should apply a modifier when the date is selected', () => {
      const container = mount(DatePickerDay, {
        propsData: {
          day: day,
          month: 4,
        },
        provide: {
          getSelected: () => day
        }
      });
      expect(container.classes('md-datepicker__day--selected')).toEqual(true);
    });

    it('should apply a modifier when the date is today`s date', () => {
      const container = mount(DatePickerDay, {
        propsData: {
          day: moment(),
          month: 4,
        },
      });
      expect(container.classes('md-datepicker__day--today')).toEqual(true);
    });

    it('should apply a modifier when the date is in focus', () => {
      const container = mount(DatePickerDay, {
        propsData: {
          day: day,
          month: 4,
        },
        provide: {
          getFocus: () => day
        }
      });
      expect(container.classes('md-datepicker__day--focus')).toEqual(true);
    });
  });

  it('when Day the disabled, should set disabled prop on button', () => {
    const container = mount(DatePickerDay, {
      propsData: {
        day: day,
        month: 4,
        filterDate: () => true
      },
    });
    expect(container.attributes().disabled).toEqual('disabled');
  });

  it('onclick of Day should call the callback in context', () => {
    const onClickFn = jest.fn();
    const container = mount(DatePickerDay, {
      propsData: {
        day: day,
        month: 4,
      },
      provide: {
        getFocus: () => day,
        handleDayClick: onClickFn
      }
    });

    container.trigger('click');
    expect(onClickFn).toHaveBeenCalled();
  });

});

