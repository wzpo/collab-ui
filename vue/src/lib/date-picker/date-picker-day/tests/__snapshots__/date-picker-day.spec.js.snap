// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerDay should match snapshot 1`] = `
<button aria-label="Sunday, April 1st 2018" alt="Sunday, April 1st 2018" type="button" tabindex="-1" role="button" class="md-datepicker__day md-button md-button--circle md-button--28 md-datepicker__day--selected" aria-selected="true"><span class="md-button__children" style="opacity: 1;"><div aria-label="Sunday, April 1st 2018">
    1
  </div></span></button>
`;
